config/src/parameters/user.rs

```diff
@@ -471,18 +470,15 @@ pub struct Logger {
     pub level: Level,
     /// Output format
     pub format: LoggerFormat,
-    #[cfg(feature = "tokio-console")]
-    /// Address of tokio console (only available under "tokio-console" feature)
+    /// Address of tokio console
     pub tokio_console_address: SocketAddr,
```

Comment:
```suggestion
    pub tokio_console_address: Option<SocketAddr>,
```

should it not be optional now?

Reply:
It is optional for users, but since it has a default value, it is always `SocketAddr` here.

Reply:
oh, I see. Do you think it should have a default?

Reply:
I haven't dived into the purpose of this feature, to be honest. So, I don't have an opinion on that. Will check it out.

Reply:
What I have found:

- There is a broken code in `iroha_logger`, relying on obsolete name `tokio_console_addr`. Our CI doesn't detect validity of different features presets.
  https://github.com/hyperledger/iroha/blob/f3e7097c9c07dc024f6de32d968f594d284880c8/logger/src/lib.rs#L120-L132
- I think having a default value for this parameter is reasonable, because this functionality is switched with a feature flag of `iroha_logger`. However, `iroha`'s feature flags do not enable this feature of logger in any way.
- Although `tokio_console` is directly related to `tracing`, I think it would make more sense to move the parameter from `logger` to `telemetry` section of the config. I.e. not `logger.tokio_console_address`, but `telemetry.tokio_console.address`.

So, what I propose:

- Move the parameter from `logger` to `telemetry.tokio_console.address`, alongside with "regular" and "dev" telemetries
- Possibly nest "regular" telemetry from `telemetry.*` to `telemetry.substrate` or whatever
- Possibly continue feature-gating this field in `iroha_config`. As well as other telemetry-related fields, because telemetry might be totally disabled by features of `iroha`
- By continuing feature-gating these parameters, it is reasonable for them to have default values when they are enabled, imo.

Your thoughts?

Reply:
A contra point against using feature-gating in `iroha_config`: it complicates snapshot-based testing.